src/website/app/pages/RenderProps/render-props.md

```diff
@@ -0,0 +1,159 @@
+## Overview
+
+_Excerpted from the official [React docs](https://reactjs.org/docs/render-props.html)._
+
+> The term "render prop" refers to a simple technique for sharing code between
+> React components using a prop whose value is a function.
+>
+> A component with a render prop takes a function that returns a React element
+> and calls it instead of implementing its own render logic.
+>
+> ```jsx
+> <DataProvider render={data => (
+>   <h1>Hello {data.target}</h1>
+> )} />
+> ```
+>
+> _It’s important to note that just because the pattern is called
+> "render props" it does not mean that the prop will always be named render.
+> In fact, any prop that is a function that a component uses to know what to
+> render is technically a "render prop"._
+
+
+## Usage in Mineral
+
+Numerous Mineral components make use of render props in order provide greater
+customization and rendering flexibility.  e.g.
+[Table](/components/table),
+[Select](/components/select),
+[Dropdown](/components/dropdown),
+[Popover](/components/popover),
+[Menu](/components/menu)
+
+
+### When to use
+
+Render props are ideal to use when the default component implementation is
+insufficient for the unique needs of your application and when the desired
+customizations cannot be implemented by a simpler means, such as with Mineral's
+robust [styling](/styling) and/or [theming](/theming) capabilities.
+
+
+### Callback Parameters
+
+During the render cycle, the render prop callback function will be invoked with
+an object parameter containing the following information and utilities to help
+customize your implementation.
+
+* __props__: Props to be spread to your custom component
+    * Note that these may not all be necessary for your implementation, or even
+    valid DOM attributes, so you’ll want to apply them judiciously.  _If your
+    component has been created with
+    [createStyledComponent](/styling#customization-techniques-themes-createstyledcomponent),
+    invalid DOM attributes will be removed automatically._
+    * Further, you can override and/or compose these props as you see fit.
+* __state__: The internal state of the component, if any.
+* __helpers__: Miscellaneous helper methods, if any.
+
+
+### Performance Considerations
+
+#### Define render callbacks outside of render
+Use an instance method, a static class method, or a separate named function that
+is otherwise created outside of render. This prevents the creation of a new
+function on each render cycle and thus does not break `PureRender` or
```

Comment:
Did you mean PureComponent here or is PureRender a different thing? On a quick look, I could only find PureRenderMixin which is legacy.

Reply:
Wops! `PureComponent`